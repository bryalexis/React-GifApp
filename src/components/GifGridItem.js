import React from 'react'

export const GifGridItem = ({id, title, url}) => {
  return (
    <div className='gifItem animate__animated animate__backInLeft'>
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}
