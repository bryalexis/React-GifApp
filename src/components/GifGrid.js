import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    // const [images, setImages] = useState([])
    
    // useEffect( () => {
    //     getGifs(category)
    //         .then(setImages);
    // }, [category]) // Si la categor√≠a cambia se actualiza el componente

    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>
            {loading && <p className='animate__animated animate__flash'>Loading...</p>}
            <div className='gifItem-grid'> 
            {
                images.map( img => (
                    <GifGridItem key={img.id} {...img} />
                ))
            }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
