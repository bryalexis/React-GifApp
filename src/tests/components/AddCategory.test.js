import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper;
    
    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories } />);
    })

    test('Debe mostrar el contenido correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe actualizar input de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hello';
        input.simulate('change', { target: { value } });
        expect( wrapper.find('p').text().trim() ).toBe( value );
    })

    test('No debe hacer POST on submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect( setCategories ).not.toHaveBeenCalled();
    })

    test('Debe llamar a setCategories y limpiar input', () => {
        const value = 'OwO';
        wrapper.find('input').simulate('change', { target: { value }});
        wrapper.find('form').simulate('submit', { preventDefault(){} });
 
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        expect(wrapper.find('input').prop('value')).toBe('');
    })

}) 