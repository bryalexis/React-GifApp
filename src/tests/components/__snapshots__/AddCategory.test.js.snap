// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe mostrar el contenido correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Add Category
  </h2>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
